proxy/core/acceptor/threadless.py

```diff
@@ -178,13 +173,38 @@ def cleanup(self, work_id: int) -> None:
         del self.works[work_id]
         os.close(work_id)
 
-    def run_once(self) -> None:
+    def _prepare_tasks(
+            self,
+            work_by_ids: Dict[int, Tuple[Readables, Writables]],
+    ) -> Set['asyncio.Task[bool]']:
+        tasks: Set['asyncio.Task[bool]'] = set()
+        assert self.loop
+        for work_id in work_by_ids:
+            readables, writables = work_by_ids[work_id]
+            task = self.loop.create_task(
+                self.works[work_id].handle_events(
+                    readables,
+                    writables,
+                ),
+            )
+            task._work_id = work_id     # type: ignore
+            # task.set_name(work_id)
+            tasks.add(task)
+        return tasks
+
+    async def run_once(self) -> None:
         assert self.loop is not None
-        with self.selected_events() as (readables, writables):
-            if len(readables) == 0 and len(writables) == 0:
+        events, work_by_ids = await self._selected_events()
+        try:
+            if len(work_by_ids) == 0:
                 # Remove and shutdown inactive connections
                 self.cleanup_inactive()
```

Comment:
have you considered only wrapping this instruction with try/except?

Reply:
After further refactoring try/except no longer used here